python/sglang/srt/managers/scheduler.py

```diff
@@ -683,11 +694,22 @@ def handle_embedding_request(
 
         # Truncate prompts that are too long
         if len(req.origin_input_ids) >= self.max_req_input_len:
```

Comment:
qq: why do we have truncation here after checking it in TokenizerManager? @merrymercy

Reply:
TokenizerManager checks the input against the context length form `config.json`.
Sometimes, the GPU memory is not enough, so the KV cache pool size will not be enough to hold even a single request with the maximum context length.
scheduler.py will do this fine-grained check against the KV cache size pool.